src/servers/ZoneServer2016/managers/decaymanager.ts

```diff
@@ -61,6 +64,37 @@ export class DecayManager {
     }, this.decayTickInterval);
   }
 
+  private constructionGriefCheck(server: ZoneServer2016) {
+    for (const a in server._constructionFoundations) {
+      const foundation = server._constructionFoundations[a];
+      if (
+        foundation.itemDefinitionId == Items.FOUNDATION ||
+        foundation.itemDefinitionId == Items.GROUND_TAMPER
+      ) {
+        if (
+          Object.keys(foundation.occupiedWallSlots).length <=
+            this.griefCheckSlotAmount &&
+          Object.keys(foundation.occupiedShelterSlots).length == 0 &&
+          Object.keys(foundation.occupiedExpansionSlots).length == 0
+        ) {
+          if (
+            Date.now() - foundation.creationTime >=
```

Comment:
This if statement could be done before the one above so we could save cpu time since this statement is a small comparison instead of creating an array of keys to check length ( array that would be deleted after this stack also )